server/runtime/src/main/java/org/infinispan/server/configuration/security/RealmConfigurationBuilder.java

```diff
@@ -157,4 +179,35 @@ SSLContext getSSLContext() {
    public void addFeature(ServerSecurityRealm.Feature feature) {
       features.add(feature);
    }
+
+   private SecurityRealm cacheable(SecurityRealm realm) {
+      int maxEntries = attributes.attribute(CACHE_MAX_SIZE).get();
+      if (maxEntries > 0 && realm instanceof CacheableSecurityRealm) {
+         if (cache == null) {
+            cache = new LRURealmIdentityCache(maxEntries, attributes.attribute(CACHE_LIFESPAN).get());
+         }
+         if (realm instanceof ModifiableSecurityRealm) {
+            return new CachingModifiableSecurityRealm((CacheableSecurityRealm) realm, cache);
+         } else {
+            return new CachingSecurityRealm((CacheableSecurityRealm) realm, cache);
+         }
+      } else {
+         return realm;
+      }
+   }
+
+   public void addRealm(String realmName, SecurityRealm realm) {
+      addRealm(realmName, realm, null);
+   }
+
+   public void addRealm(String realmName, SecurityRealm realm, Consumer<SecurityDomain.RealmBuilder> realmBuilderConsumer) {
+      SecurityDomain.RealmBuilder realmBuilder = domainBuilder.addRealm(realmName, cacheable(realm));
+      if (realmBuilderConsumer != null) {
+         realmBuilderConsumer.accept(realmBuilder);
+      }
+      realmBuilder.build();
```

Comment:
I know it's not our API but I feel like it merits a comment, seeing the result of `build()` unused is puzzling.